img/CacheIndexing.svg

```diff
@@ -0,0 +1,4 @@
.5" y="30" width="120" height="30" fill="none" stroke="none" pointer-events="all"/><g fill="rgb(0, 0, 0)" font-family="Helvetica" text-anchor="middle" font-size="12px"><text x="161" y="49.5">Tag bits</text></g><rect x="440" y="131" width="30" height="10" fill="#b3b3b3" stroke="rgb(0, 0, 0)" pointer-events="all"/><path d="M 440 136 Q 400 131 395 196 Q 390 261 306.37 261" fill="none" stroke="rgb(0, 0, 0)" stroke-miterlimit="10" pointer-events="stroke"/><path d="M 301.12 261 L 308.12 257.5 L 306.37 261 L 308.12 264.5 Z" fill="rgb(0, 0, 0)" stroke="rgb(0, 0, 0)" stroke-miterlimit="10" pointer-events="all"/><rect x="70" y="246" width="230" height="30" fill="none" stroke="none" pointer-events="all"/><g fill="rgb(0, 0, 0)" font-family="Helvetica" text-anchor="end" font-size="12px"><text x="297.5" y="265.5">Tag bits match (0b1001000010......0001)</text></g><rect x="570" y="45" width="120" height="30" fill="none" stroke="none" pointer-events="all"/><g fill="rgb(0, 0, 0)" font-family="Helvetica" font-size="12px"><text x="571.5" y="64.5">within the cache line</text></g><rect x="440" y="151" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="440" y="111" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="440" y="211" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="280" y="111" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="280" y="131" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="280" y="151" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="280" y="211" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="180" y="111" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="180" y="131" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="180" y="151" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="180" y="211" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="80" y="111" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="80" y="131" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="80" y="151" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="80" y="211" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="600" y="111" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="600" y="131" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="600" y="151" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/><rect x="600" y="211" width="30" height="10" fill="none" stroke="rgb(0, 0, 0)" pointer-events="all"/></g></svg>
```

Comment:
figure looks good, I think the grouping curly brace for the offset bits might be a bit skewed though? I tried two different online SVG viewers and they both rendered the same, but might be an error on my side.

I'm guessing there are two bits missing in the offset portion of the `address`, as only four bits are grouped by the curly brace but there are six bits pointed to by the arrow?

Reply:
I'd suggest explicitly stating the cache line size, total cache size, set associativity etc for this example, since it would help make the example clearer, and some of these values are already implicitly fixed in the figure. E.g. 6 offset bits for a 64-byte sized cache line, assuming byte addressing, and 6 set bits for a 4-way associative cache that has 64 sets, for a total size of 4 * 64 * 64 = 16KB.

Reply:
@JLouisKaplan-Arm : For the curly brace looking a bit skewed: I know!
The problem is that in the PDF vs the HTML, the curly braces come out in different locations. Or rather, the curly braces do come out in the same locations in the PDF and HTML, but the font that ends up being used for the bit sequence in the PDF vs HTML versions seems to be having different character pixel widths.
By changing to font to Courier New (a mono-spaced font), the difference is a lot smaller than it was when using a non-monospaced font. But the issue is still there.
I'm afraid I haven't found a way so far to make fonts identical between the PDF and HTML versions of the SVG diagrams.... I thought it was best to not block progress on this PR because of this issue now that the curly braces look acceptable (albeit far from perfect).

@g-kouv : yes, good suggesting for explicitly stating cache line size, cache size etc on the figure; I'll follow your suggestion.

Reply:
ack, that sounds like a fiddly issue. Sounds like a good plan to progress with the PR for now.